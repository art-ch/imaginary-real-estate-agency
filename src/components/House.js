import React, { useEffect } from 'react';
import 'bootstrap/dist/css/bootstrap.min.css';

import HouseWrapper from '../styled/House';
import { Container, Row, Col, Image, Button, Spinner } from 'react-bootstrap';

import { useDispatch, useSelector } from 'react-redux';
import { useParams, useNavigate } from 'react-router-dom';
import { getSingleHouse, resetSingleHouse } from '../redux/ducks/getHouse';
import { priceHandler } from '../utils';

const House = () => {
  const { id } = useParams();
  const dispatch = useDispatch();
  const navigate = useNavigate();

  useEffect(() => {
    dispatch(getSingleHouse(id));
    return () => {
      dispatch(resetSingleHouse(id));
    };
    // eslint-disable-next-line
  }, []);

  const currentHouse = useSelector((state) => state.getHouseReducer.house);
  const { house, gallery } = currentHouse;

  if (house && gallery) {
    const { image, title, price, address, description, name } = house;
    return (
      <HouseWrapper>
        <Container className="img-container">
          <Image src={image} alt="" />
        </Container>
        <Container className="my-2 g-4">
          <h1>{title}</h1>
          <p className="h4">{priceHandler(price)}</p>
          <p className="h6">{address}</p>
          <p>{description}</p>
          <p className="text-uppercase fst-italic">{name}</p>
          <Container className="container-sm p-0 mb-3">
            <Row className="flex-column g-2">
              <Col className="w-100 d-flex">
                <Button variant="outline-primary" className="me-2 w-100">
                  Contact Agent
                </Button>
                <Button className="w-100">Take a Tour</Button>
              </Col>
              <Col>
                <Button
                  className="w-100"
                  variant="warning"
                  onClick={() => navigate('/')}
                >
                  Back
                </Button>
              </Col>
            </Row>
          </Container>
          <Container className="p-0">
            <Row>
              <h2 className="text-center">Gallery</h2>
              {gallery.map((image, id) => {
                return (
                  <Col className="mb-3 col-12 col-sm-4 col-xl-3" key={id}>
                    <a href={image}>
                      <img src={image} alt="" />
                    </a>
                  </Col>
                );
              })}
            </Row>
          </Container>
        </Container>
      </HouseWrapper>
    );
  }
  return (
    <div className="mt-3 text-center">
      <Spinner className="center-block" animation="border" variant="primary" />
    </div>
  );
};

export default House;
