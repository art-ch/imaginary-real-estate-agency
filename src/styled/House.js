import styled from 'styled-components';

const HouseWrapper = styled.main`
  margin: 0 auto;
  .container {
    max-width: 1000px;
  }
`;

export default HouseWrapper;
