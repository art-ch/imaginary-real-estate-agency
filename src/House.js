import React, { useContext } from 'react';
import { Context } from './Context';
import styled from 'styled-components';
import { Container, Image, Button, Carousel } from 'react-bootstrap';
import 'bootstrap/dist/css/bootstrap.min.css';

const House = () => {
  const gallery = [
    'https://images.pexels.com/photos/164338/pexels-photo-164338.jpeg?auto=compress&cs=tinysrgb&dpr=2&h=750&w=1260',
    'https://images.pexels.com/photos/1457842/pexels-photo-1457842.jpeg?auto=compress&cs=tinysrgb&dpr=2&h=750&w=1260',
    'https://images.pexels.com/photos/1080721/pexels-photo-1080721.jpeg?auto=compress&cs=tinysrgb&dpr=2&h=750&w=1260',
    'https://images.pexels.com/photos/90317/pexels-photo-90317.jpeg?auto=compress&cs=tinysrgb&dpr=2&h=750&w=1260',
    'https://images.pexels.com/photos/1454804/pexels-photo-1454804.jpeg?auto=compress&cs=tinysrgb&dpr=2&h=750&w=1260',
    'https://images.pexels.com/photos/5252546/pexels-photo-5252546.jpeg?auto=compress&cs=tinysrgb&dpr=2&h=750&w=1260',
    'https://images.pexels.com/photos/13975/pexels-photo-13975.jpeg?auto=compress&cs=tinysrgb&dpr=2&h=750&w=1260',
  ];
  const { priceHandler } = useContext(Context);
  return (
    <Wrapper>
      <Container className="img-container">
        <Image
          src="https://images.pexels.com/photos/164338/pexels-photo-164338.jpeg?auto=compress&cs=tinysrgb&dpr=2&h=750&w=1260"
          alt=""
        />
      </Container>
      <Container className="my-2 g-4">
        <h1>Townhome</h1>
        <p className="h4">{priceHandler(289900)}</p>
        <p className="h6">1933 Frum Street, Nashville, TN 37212</p>
        <p>
          Broker ipsum dolor sit amet, consectetuer stainless steel elit. Trendy
          shops commodo ligula eget dolor. Cum sociis luxury penatibus et magnis
          dis modern interior montes, nascetur soaring views mus. Location
          location location quam felis, ultricies nec, pellentesque
          rent-controlled tenant in place, pretium quis, sem.
        </p>
        <Carousel className="my-5">
          {gallery.map((image) => {
            return (
              <Carousel.Item>
                <Container className="img-container">
                  <img className="w-100 h-100" src={image} alt="" />
                </Container>
              </Carousel.Item>
            );
          })}
        </Carousel>
        <p className="text-uppercase fst-italic">Tiffany Marin</p>
        <Button>Contact Agent</Button>
      </Container>
    </Wrapper>
  );
};

export default House;

const Wrapper = styled.main`
  margin: 0 auto;
  .img-container {
    height: 300px;
    img {
      width: 100%;
      height: 100%;
      object-fit: cover;
    }
  }
  .container {
    max-width: 1000px;
  }

  @media (min-width: 767px) {
    .img-container {
      height: 400px;
    }
  }

  @media (min-width: 1200px) {
    .img-container {
      height: 570px;
    }
  }
`;
