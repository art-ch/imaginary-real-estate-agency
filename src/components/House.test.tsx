import { Provider } from 'react-redux';
import { BrowserRouter, Routes, Route } from 'react-router-dom';
import { mount } from 'enzyme';

import House from './House';
import store from '../redux/configureStore';

const component = mount(
  <Provider store={store}>
    <BrowserRouter>
      <Routes>
        <Route path="/houses/house/7" element={<House />} />
      </Routes>
    </BrowserRouter>
  </Provider>
);

describe('Component', () => {
  it('should match snapshot', () => {
    expect(component).toMatchSnapshot();
  });
});
