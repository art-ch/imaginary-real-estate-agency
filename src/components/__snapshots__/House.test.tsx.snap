// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component should match snapshot 1`] = `
<Provider
  store={
    Object {
      "@@observable": [Function],
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <BrowserRouter>
    <Router
      location={
        Object {
          "hash": "",
          "key": "default",
          "pathname": "/",
          "search": "",
          "state": null,
        }
      }
      navigationType="POP"
      navigator={
        Object {
          "action": "POP",
          "back": [Function],
          "block": [Function],
          "createHref": [Function],
          "forward": [Function],
          "go": [Function],
          "listen": [Function],
          "location": Object {
            "hash": "",
            "key": "default",
            "pathname": "/",
            "search": "",
            "state": null,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <Routes />
    </Router>
  </BrowserRouter>
</Provider>
`;
